### PR TITLE
plugin: consul: added more statistics for server and services

### PR DESCRIPTION
**Pull request by harnash**
### Required for all PRs:

- [ ] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

As requested in #913 consul plugin now returns some basic stats about
the server (peers, nodes, services, leadership). It is also possible to
query all service health registered in the catalog but since it is quite
slow it is set as optional.

I've tested it and it does gather service health checks sloooow - no wonder since it has to query each service separately for it's health checks. So if I understand correctly this options would be used when one wants to query only Consul servers and leave consul agents/nodes intact. In other scenario one would setup consul to query all consul nodes including servers and health checks should be gathered in `consul_health_checks` with proper tag (`service_name`).

@oryband please take a look at it and tell me if output looks OK to you.

### CODE REVIEW

File: plugins/inputs/consul/consul.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"net/http"
 
 	"github.com/hashicorp/consul/api"
+	consulStruct "github.com/hashicorp/consul/consul/structs"
```
Comment (sparrc) [inline plugins/inputs/consul/consul.go]: why did you name this? what is the actual package name?

Comment (sparrc) [inline plugins/inputs/consul/consul.go]: is this meant to be an exported type? I can't find any documentation that hashicorp meant for this to be used to extract metrics.

Comment (sparrc) [inline plugins/inputs/consul/consul.go]: can this be used?: https://godoc.org/github.com/hashicorp/consul/api

Comment (harnash) [inline plugins/inputs/consul/consul.go]: @sparrc I'll look at this during the weekend

**Comment by harnash** [comment]
@sparrc I've updated Consul library and that simplified dependencies (I was using older one and it didn't have constants defined in the `api` package). I'm in process of writing tests to cover new functionality.

**Comment by harnash** [comment]
@sparrc I think now it should be OK

**Comment by sparrc** [comment]
@johnrengelman could you please review?

**Comment by glinton** [comment]
@harnash do you think you could update your branch (merge or rebase master) before we get to a full review? Thanks and sorry this has taken so long.

**Comment by harnash** [comment]
Yeah, it's been a while since I wrote this. I will be able to take a look next week when I get back from holidays 😄

**Comment by harnash** [comment]
OK. I've checked this PR and this are my thoughts.
1. Original repository I was working on is removed (probably due to some cleanups) and I would need to redo all the work done here. Not huge task but still.
2. Our company do not need this functionality anymore so it will be a bit difficult right now to properly test it on larger scale to make sure we do not introduce high cardinality problems. Additionally maintaining it would be difficult since our infrastructure looks quite different from the times when this PR was created.

I vote to close this PR for now and if anyone believe it is needed feature I could try to resurrect it.

@glinton what do you think?

**Comment by danielnelson** [comment]
Thanks for the update @harnash, very understandable.  Github still has the code here, so it should always be possible for someone to revive it from here.

**Pull request closed without merging**